src/main/java/com/redhat/devtools/lsp4ij/client/LanguageClientImpl.java

```diff
@@ -164,13 +165,52 @@ public void notifyProgress(ProgressParams params) {
     }
 
     @Override
-    public void dispose() {
-        this.disposed = true;
-        this.progressManager.dispose();
+    public CompletableFuture<List<Object>> configuration(ConfigurationParams params) {
+        return CompletableFuture.supplyAsync(() -> {
+            // See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration
+            List<Object> settings = new ArrayList<>();
+            for (ConfigurationItem item : params.getItems()) {
+                String section = item.getSection();
+                Object result = findSettings(section);
+                // The result is the configuration setting or null, according to the spec:
+                //  - If a scope URI is provided the client should return the setting scoped to the provided resource.
+                //  - If the client can’t provide a configuration setting for a given scope then null needs to be present in the returned array.
+                settings.add(result);
+            }
+            return settings;
+        });
     }
 
-    public boolean isDisposed() {
-        return disposed || getProject().isDisposed();
+    /**
+     * Returns the settings retrieved by the given section and null otherwise.
+     *
+     * <p>
+     * This method is implemented by default to work with JsonObject ({@link LanguageClientImpl#createSettings()}
+     * must return an instance of {@link JsonObject} but this method can be override it to work with another structure.
```

Comment:
can be overridden to return another structure

Reply:
fixed